storage/pebble/registers.go

```diff
@@ -134,6 +137,12 @@ func (s *Registers) Store(
 	}
 	s.latestHeight.Store(height)
 
+	// TODO: This is temporary solution, as I think, this should be changed when register DB pruner will be implemented.
+	//       The pruner responsibility is to change first height of the node, base on pruning result. Issue https://github.com/onflow/flow-go/issues/6068
```

Comment:
the pruner should update the first height value stored in the db. this makes sure if the node crashes/restarts mid-prune, the db state is still consistent.

The firstHeight we store here is used for runtime checks and can be managed based on the threshold. This ensures requests see consistent behavior even if pruning is done infrequently. 

When we add the pruner, it should still call back to update `firstHeight`, but we would only do the update if the new value is higher than the cached value